### PR TITLE
Implement Redux

### LINKED ISSUE #60: Getting Started with Redux
**Is your feature request related to a problem? Please describe.**
Help manage the myPokeList state.

### PR DESCRIPTION
**Pull request by krferrell**
## Changes
_List Changes Introduced by this PR_
1. Moved allPokemon state and functionality into Redux 
2. Refactored the way pokemon that the user owns is rendered

## Purpose
Makes global state management easier

## Approach
Asked Kendrick for help

Closes #60

### CODE REVIEW

**Review by gonzalespaulb — APPROVED** [review]
Big ups to @kmausolf for the help.

**Review by JayDampitan — APPROVED** [review]
Like what my guy Eli said. We Love you Kenny!

File: pokeapp/src/App.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import React from "react";
 import { Brain } from "./Brain";
+import store from "./redux/store";
+import { Provider } from "react-redux";
 
 const App = () => {
   return (
-    <div className="App">
+    <Provider store={store}>
       <Brain />
```
Comment (BlackCubes) [inline pokeapp/src/App.js:10]: Big brain

File: pokeapp/src/Brain.js [hunk 2]
```diff
@@ -136,49 +137,49 @@ export const Brain = () => {
...
+  //   }
+  // };
+
+  // const subtractPokeDollars = (pokemon) => {
+  //   setPokeDollars((currentAmount) => currentAmount - pokemon.value);
+  // };
+
+  // const addPokeDollars = (pokemon) => {
+  //   setPokeDollars((currentAmount) => currentAmount + pokemon.value);
+  // };
```
Comment (Lap343) [inline pokeapp/src/Brain.js:182]: do you still need this?

**Review by Lap343 — COMMENTED** [review]
Woo! Really man great job implementing redux into your guys app. There is just one giant block of commented code

File: pokeapp/src/Brain.js [hunk 3]
```diff
@@ -57,7 +58,7 @@ export const Brain = () => {
     //Enrich each object with custom properties
     const updatedPokemonList = updatePokeInfo(resolvedList, weaknessInfo);
     //Save the completed list and complete initialization
-    setAllPokemon(updatedPokemonList);
+    dispatch(setPokeList(updatedPokemonList));
```
Comment (BlackCubes) [inline pokeapp/src/Brain.js:61]: Nice redux implementation!

File: pokeapp/src/Brain.js (hunk 2 above)
Comment (kmausolf) [inline pokeapp/src/Brain.js:182]: We can remove this now.

**Review by JorgeLVilla — APPROVED** [review]
Great job with Redux!

File: pokeapp/src/Brain.js (hunk 2 above)
Comment (BlackCubes) [inline pokeapp/src/Brain.js:182]: Beastly

File: pokeapp/src/Brain.js [hunk 4]
```diff
@@ -192,46 +193,29 @@ export const Brain = () => {
...
-              setIsMoreInfo={setIsMoreInfo}
-            />}
-
+              <SideBar
+                selectedPokemon={selectedPokemon}
+                makeUpperCase={makeUpperCase}
+                isMoreInfo={isMoreInfo}
+                setIsMoreInfo={setIsMoreInfo}
+              />
+            }
```
Comment (BlackCubes) [inline pokeapp/src/Brain.js:166]: Looks cleaner

**Review by BlackCubes — APPROVED** [review]
Bruh, dis be lookin real GEWD! Looks clean and well done. You truly are a beast. Good job!
kobe

File: pokeapp/src/redux/store.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  reducer: {
+    //setting the PokemonSlice component to have a name of pokemon in the store
+    //when we use the useSelector hook, we call state.pokemon to access pokemonSlice
+    pokemon: pokemonSlice,
+  },
+  middleware: (getDefaultMiddleware) =>
+    getDefaultMiddleware({
+      thunk: {},
+      serializableCheck: false,
+    }),
```
Comment (kmausolf) [inline pokeapp/src/redux/store.js:14]: It's so easy to add thunk now.  I love that they included it in the defaults!

**Pull request merged**